src/tests/Microsoft.PowerFx.Interpreter.Tests/DatabaseSimulationTests.cs

```diff
@@ -44,16 +47,36 @@ public void DatabaseSimulation_Test(string expr, bool checkSuccess)
             }
 
             IExpressionEvaluator run = check.GetEvaluator();
-            FormulaValue result = run.EvalAsync(CancellationToken.None, runtimeConfig).Result;
 
-            Assert.IsType<BlankValue>(result);
+            using (var cts = new CancellationTokenSource(1500))
+            {
+                var aggregateExceptionFired = false;
+
+                try
+                {
+                    FormulaValue result = run.EvalAsync(cts.Token, runtimeConfig).Result;
+                    Assert.IsType<InMemoryRecordValue>(result);
+                }
+                catch (AggregateException agg)
+                {
+                    Assert.True(expectTaskCancalledException);
+                    Assert.IsType<TaskCanceledException>(agg.InnerException);
+                    aggregateExceptionFired = true;
+                }
+                finally
+                {
+                    Assert.True(!expectTaskCancalledException ^ aggregateExceptionFired);
```

Comment:
>!expectTaskCancalledException ^ aggregateExceptionFired [](http://example.com/codeflow?start=32&length=55)

`expectTaskCancelledException == aggregateExceptionFired` is easier to read than using a xor

Reply:
:)